samples/csharp/getting-started/ONNXExport/ONNXExport/ONNXExport/Program.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace ONNXExport
+{
+    class Program
+    {
+        private static string currentDirectory = AppDomain.CurrentDomain.BaseDirectory;
+        private static string TRAIN_DATA_FILEPATH = Path.Combine(currentDirectory, @"..\..\..\..\..\..\Regression_TaxiFarePrediction\TaxiFarePrediction\Data\taxi-fare-train.csv");
+        private static string TEST_DATA_FILEPATH = Path.Combine(currentDirectory, @"..\..\..\..\..\..\Regression_TaxiFarePrediction\TaxiFarePrediction\Data\taxi-fare-test.csv");
+
+        static void Main(string[] args)
+        {
+            var mlContext = new MLContext();
+
+            // Load training data
+            IDataView trainingDataView = mlContext.Data.LoadFromTextFile<ModelInput>(
+                                            path: TRAIN_DATA_FILEPATH,
+                                            hasHeader: true,
+                                            separatorChar: ',');
+
+            // Load test data
+            IDataView testDataView = mlContext.Data.LoadFromTextFile<ModelInput>(
+                                            path: TEST_DATA_FILEPATH,
+                                            hasHeader: true,
+                                            separatorChar: ',');
+
+
+            // Create data processing pipeline for training ML.NET model
+            var dataProcessPipeline = mlContext.Transforms.Categorical.OneHotEncoding(new[] { new InputOutputColumnPair("vendor_id", "vendor_id"), new InputOutputColumnPair("payment_type", "payment_type") })
+                                      .Append(mlContext.Transforms.Concatenate("Features", new[] { "vendor_id", "payment_type", "rate_code", "passenger_count", "trip_time_in_secs", "trip_distance" }));
+
+            // Set training algorithm and append to pipeline
+            var trainer = mlContext.Regression.Trainers.Sdca(labelColumnName: "fare_amount", featureColumnName: "Features");
+
+            var trainingPipeline = dataProcessPipeline.Append(trainer);
+
+            // Train ML.NET model on training data
+            ITransformer model = trainingPipeline.Fit(trainingDataView);
+
+            // You need a transformer and input data to convert an ML.NET model to an ONNX model
+            // By default, the ONNX conversion will generate the ONNX file with the latest OpSet version
+            using (var stream = File.Create("taxi-fare-model.onnx"))
+                mlContext.Model.ConvertToOnnx(model, trainingDataView, stream);
```

Comment:
Do you think we should talk about how they can pick which columns to output from the ONNX model? Or is that more advanced than this tutorial should cover?

Reply:
What would be the use case for this and is there an example I can look at to bring over?

Reply:
So we added it in in [1.5.4](https://github.com/dotnet/machinelearning/releases/tag/v1.5.4). Its gives the user more performance. By default, since we don't know which columns the user wants, we are outputting all of the columns from the onnx model. This works great, but makes the model bigger and decreases performance. For many situations that wont matter, but in cases where they do need good performance it can make a very large difference.